<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>620</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>100</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Input an array</string>
    </property>
   </widget>
   <widget class="QScrollArea" name="scrollArea">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>600</width>
      <height>60</height>
     </rect>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>598</width>
       <height>58</height>
      </rect>
     </property>
     <widget class="QTextEdit" name="inputArrayEdit">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>600</width>
        <height>60</height>
       </rect>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QPushButton" name="inputByFileBtn">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>10</y>
      <width>80</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Input by file</string>
    </property>
   </widget>
   <widget class="QPushButton" name="generateInputBtn">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>120</y>
      <width>110</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Generate input</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>120</y>
      <width>70</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Input size</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="inputSizeEdit">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>120</y>
      <width>90</width>
      <height>30</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>120</y>
      <width>121</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Choose generate type</string>
    </property>
   </widget>
   <widget class="QComboBox" name="generateTypeBox">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>120</y>
      <width>120</width>
      <height>30</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Ascending</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Descending</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Random</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="inputArrayBtn">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>160</y>
      <width>125</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Input this array</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>200</y>
      <width>100</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Input search value</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="searchValueEdit">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>200</y>
      <width>461</width>
      <height>30</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>240</y>
      <width>131</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Choose search algorithm</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="linearSearchCheck">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>240</y>
      <width>80</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Linear</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="binarySearchCheck">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>240</y>
      <width>80</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Binary</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="ternarySearchCheck">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>240</y>
      <width>80</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Ternary</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="jumpSearchCheck">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>240</y>
      <width>80</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Jump</string>
    </property>
   </widget>
   <widget class="QPushButton" name="search">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>320</y>
      <width>125</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Search</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="exponentialSearchCheck">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>280</y>
      <width>80</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Exponetial</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="interpolationSearchCheck">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>280</y>
      <width>80</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Interpolation</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="hashingSearchCheck">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>280</y>
      <width>80</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Hashing</string>
    </property>
   </widget>
   <widget class="QScrollArea" name="scrollForLog">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>380</y>
      <width>600</width>
      <height>180</height>
     </rect>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents_4">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>598</width>
       <height>178</height>
      </rect>
     </property>
     <widget class="QTextEdit" name="resultLog">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>600</width>
        <height>180</height>
       </rect>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>620</width>
     <height>17</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
